src/main/python/mainQt.py

```diff
@@ -225,10 +225,11 @@ def reset_ui_record_number(self):
         self.ui.horizontalSlider.setValue(0)
 
     def open_dialog(self) -> None:
-        """
-        TODO : To be implemented
-        """
-        pass
+        fileName, _ = QFileDialog.getOpenFileName(
+            self, "Open record", "", "Sim Replay files (*.sr);;All files (*.*)"
+        )
+        if fileName:
+            self._model.open_file(fileName)
```

Comment:
UI of the slider bar shall be updated according to the new table loaded (same lines as line 239 to 242)